src/Setup/DevDivInsertionFiles/DevDivInsertionFiles.csproj

```diff
@@ -142,6 +142,7 @@
       <_Dependency Remove="Nerdbank.Streams"/>
       <_Dependency Remove="Newtonsoft.Json"/>
       <_Dependency Remove="NuGet.VisualStudio"/>
+      <_Dependency Remove="NuGet.VisualStudio.Contracts"/>
```

Comment:
Why is this change is needed?

Reply:
`NuGet.VisualStudio` contains only `COMImportAttribute` interfaces that can be embedded via the "embed interop types" compiler flag, so that `NuGet.VisualStudio.dll` is not loaded at runtime.

That's not possible with Visual Studio's brokered service architecture, so we created a new assembly, and therefore new package, instead.

Reply:
I'm not certain.  @nkolev92 @jasonmalinowski ?

Reply:
how is the brokered service arch involved here? I thought this was entirely in-process. Why is this something that is exposed to the API consumer?

Reply:
These new NuGet APIs are implemented as brokered services so that they work in live share and codespaces scenarios, in addition to in-process. My understanding is that COM interfaces don't support `Task<T>`, so even if we didn't use brokered services, we still couldn't use COM interop types and therefore in order to keep `NuGet.VisualStudio` as a pure interop assembly that can be embedded and not shipped by components using it, we'd still need a new assembly.

As for in-process, my understanding is yes, but that's a question for the team that implements the service broker. For what it's worth, the [best practises](https://dev.azure.com/devdiv/DevDiv/_git/VS.RPC.Contracts?path=%2Fdoc%2Fbestpractices.md&_a=preview) document says:

> appears on an interface for a service which may sometimes be running in the same process as its client

so it certainly gives the impression that the service broker stays in-process when it can. But NuGet just exposes our interface via the brokered service client, so it's up to that client's implementation on what to do, it's out of NuGet's control (except for not using this infrastructure).

Reply:
What is the expected pattern for a third-pary VSIX to consume these APIs?